BASE/src/Microsoft.ApplicationInsights/Channel/ITelemetryChannel.cs

```diff
@@ -31,5 +32,11 @@ public interface ITelemetryChannel : IDisposable
         /// Flushes the in-memory buffer.
         /// </summary>
         void Flush();
+
+        /// <summary>
+        /// Flushes the in-memory buffer asynchronously.
+        /// </summary>
+        /// <returns>The task to await.</returns>
+        Task<bool> FlushAsync(CancellationToken cancellationToken);
```

Comment:
adding method to the interface is a [breaking change](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/breaking-changes#binary-breaking-change) and needs major version upgrade.

Reply:
Perhaps it could/should be in a new interface. e.g. `ITelemetryChannel2` or `IAsyncFlushable`